From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: TheKit <nekit1000@gmail.com>
Date: Tue, 28 Sep 2021 19:28:31 +0300
Subject: [PATCH] (halium) fs_mgr: ignore ro flag for logical partitions so
 they can be written

Allows to install OTA updates to rootfs on system partition in UBPorts recovery.

Change-Id: Ie3f3332c5a32a0a4b0ac595caa08412386b6b5a2
---
 fs_mgr/fs_mgr_dm_linear.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/fs_mgr/fs_mgr_dm_linear.cpp b/fs_mgr/fs_mgr_dm_linear.cpp
index ee6ffdb18..7eb13d1a0 100644
--- a/fs_mgr/fs_mgr_dm_linear.cpp
+++ b/fs_mgr/fs_mgr_dm_linear.cpp
@@ -160,7 +160,8 @@ bool CreateLogicalPartitions(const LpMetadata& metadata, const std::string& supe
             continue;
         }
         std::string path;
-        if (!CreateLogicalPartition(metadata, partition, false, {}, super_device, &path)) {
+        // Halium: make sure we can write to all the partitions in recovery
+        if (!CreateLogicalPartition(metadata, partition, true, {}, super_device, &path)) {
             LERROR << "Could not create logical partition: " << GetPartitionName(partition);
             return false;
         }
-- 
2.33.0

